docs/references/api/swagger.json

```diff
@@ -506,6 +502,41 @@
         }
       }
     },
+    "/namespaces/{Namespace}/applications/{App}/from-container": {
```

Comment:
I think this endpoint is not necessary. The other 2 types of application push, set the origin implicitly in code upload or when importing from git. We could set the origin to "app from container" if the deploy endpoint is called and origin hasn't been set yet.

That said, with or without the above, we have some strange situations. E.g.

- Client pushes an application with an existing image -> origin is now set to "container"
- User uploads code for the same application -> origin is now set to "path"
The state of the application now is inconsistent, the origin is set to path but the running application was never staged with that code.

similar situations can be reached with my suggestion above. Ideally, the endpoint that should set the application's origin, is the "deploy" endpoint. Because the origin should always match the currently running image. Now how do we keep track of where an image came from? One idea is to use the image tag. If it exists and if it matches the latest PipelineRun for that application then we know we staged some code. But we still don't know where that code came from (git or path?). Since the only input to the `deploy` endpoint is the image itself, it would be nice if that image had all the information we need. But even if it did, the `deploy` endpoint can't inspect the image because it doesn't even have credentials to pull it. Maybe a good trade-off is to trust the client and accept additional parameters on the `deploy` endpoint. As in: "Hey, deploy this image and trust me, it was built from source pulled from a git remote" or "Hey, deploy this image and trust me, it's coming from a remote git". This requires the client to keep track of how the image was built and make sure to provide that information to the `deploy` endpoint. A bad client could mess up things but that's always the case right?

Reply:
Agreed. This is a larger rewrite of the PR ... It also means that the test changes can be dropped, mostly.

Reply:
Done with commit 75a1bc85515b0e6c9667064a5b19cad8f6790f97 and predecessors (included a rebase to the new main without magic domain).